File: src/main/java/ua/softgroup/matrix/server/supervisor/producer/json/TimeResponse.java
package ua.softgroup.matrix.server.supervisor.producer.json;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
public abstract class TimeResponse {

    protected int totalWorkTimeSeconds;
    protected int totalIdleTimeSeconds;
    protected double totalIdlePercentage;

    public int getTotalWorkTimeSeconds() {
        return totalWorkTimeSeconds;
    }

    public void setTotalWorkTimeSeconds(int totalWorkTimeSeconds) {
        this.totalWorkTimeSeconds = totalWorkTimeSeconds;
    }

    public int getTotalIdleTimeSeconds() {
        return totalIdleTimeSeconds;
    }

    public void setTotalIdleTimeSeconds(int totalIdleTimeSeconds) {
        this.totalIdleTimeSeconds = totalIdleTimeSeconds;
    }

    public double getTotalIdlePercentage() {
        return totalIdlePercentage;
    }

    public void setTotalIdlePercentage(double totalIdlePercentage) {
        this.totalIdlePercentage = totalIdlePercentage;
    }
}

File: src/main/scala/ua/softgroup/matrix/server/supervisor/producer/json/TimeResponse.scala
package ua.softgroup.matrix.server.supervisor.producer.json

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
abstract class TimeResponse {

  @BeanProperty
  protected var totalWorkTimeSeconds: Int = _

  @BeanProperty
  protected var totalIdleTimeSeconds: Int = _

  @BeanProperty
  protected var totalIdlePercentage: Double = _

}
